libraries/Microsoft.Bot.Builder.Azure/Storage/Blobs.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Azure.Storage
+{
+    /// <summary>
+    /// Implements <see cref="IStorage"/> using Azure Storage Blobs.
+    /// </summary>
+    /// <remarks>
+    /// This class uses a single Azure Storage Blob Container.
+    /// Each entity or <see cref="IStoreItem"/> is serialized into a JSON string and stored in an individual text blob.
+    /// Each blob is named after the store item key,  which is encoded so that it conforms a valid blob name.
+    /// If an entity is an <see cref="IStoreItem"/>, the storage object will set the entity's <see cref="IStoreItem.ETag"/>
+    /// property value to the blob's ETag upon read. Afterward, an <see cref="BlobRequestConditions"/> with the ETag value
+    /// will be generated during Write. New entities start with a null ETag.
+    /// </remarks>
+    public class Blobs : IStorage
+    {
+        private static readonly JsonSerializer JsonSerializer = JsonSerializer.Create(new JsonSerializerSettings
+        {
+            // we use All so that we get typed roundtrip out of storage, but we don't use validation because we don't know what types are valid
+            TypeNameHandling = TypeNameHandling.All,
```

Comment:
>TypeNameHandling.Al [](start = 31, length = 19)

Serializing the full .net type may cause issue (for non .net reader trying to read the data and if for anything reason you change the namespaces of an object), wouldn't it be safer to serialize just json? Or we really need full dotnet types

Reply:
I do agree.  However, our dialog stack currently requires this.